src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx

```diff
@@ -0,0 +1,174 @@
+    .reduce((prev, [_, signature]) => {
+      return prev + signature.slice(2)
+    }, '0x')
+}
+
+const getMultiSendTxs = (
+  txs: TransactionDetails[],
+  chain: ChainInfo,
+  safeAddress: string,
+  safeVersion: string,
+): MetaTransactionData[] => {
+  const safeContractInstance = getGnosisSafeContractInstance(chain, safeVersion)
+
+  return txs
+    .map((tx) => {
+      if (!isMultisigExecutionDetails(tx.detailedExecutionInfo)) return
+
+      const args = extractTxInfo(tx, safeAddress)
+      const sigs = getSignatures(args.signatures)
+
+      const data = safeContractInstance.interface.encodeFunctionData('execTransaction', [
+        args.txParams.to,
+        args.txParams.value,
+        args.txParams.data,
+        args.txParams.operation,
+        args.txParams.safeTxGas,
+        args.txParams.baseGas,
+        args.txParams.gasPrice,
+        args.txParams.gasToken,
+        args.txParams.refundReceiver,
+        sigs,
+      ])
+
+      return {
+        operation: OperationType.Call,
+        to: safeAddress,
+        value: '0',
+        data,
+      }
+    })
+    .filter(Boolean) as MetaTransactionData[]
+}
+
+const getTxsWithDetails = (txs: Transaction[], chainId: string) => {
+  return Promise.all(
+    txs.map(async (tx) => {
+      return await getTransactionDetails(chainId, tx.transaction.id)
+    }),
+  )
+}
+
+const ReviewBatchExecute = ({ data, onSubmit }: { data: BatchExecuteData; onSubmit: (data: null) => void }) => {
+  const [multiSendContract, setMultiSendContract] = useState<Multi_send_call_only>()
+  const [multiSendTxData, setMultiSendTxData] = useState<string>()
+  const [submitError, setSubmitError] = useState<Error | undefined>()
+  const chain = useCurrentChain()
+  const { safe } = useSafeInfo()
+  const provider = useWeb3()
+
+  const [txsWithDetails, error, loading] = useAsync<TransactionDetails[]>(() => {
+    if (!chain?.chainId) return
+
+    return getTxsWithDetails(data.txs, chain.chainId)
+  }, [data.txs, chain?.chainId])
+
+  useEffect(() => {
+    if (!chain) return
+
+    setMultiSendContract(getMultiSendCallOnlyContractInstance(chain.chainId))
+  }, [chain])
+
+  useEffect(() => {
+    if (!txsWithDetails || !chain) return
+
+    const multiSendTxs = getMultiSendTxs(txsWithDetails, chain, safe.address.value, safe.version)
+    const data = encodeMultiSendData(multiSendTxs)
+
+    setMultiSendTxData(data)
+  }, [chain, safe.address.value, safe.version, txsWithDetails])
+
+  const sendTx = async () => {
+    setSubmitError(undefined)
+    if (!provider || !multiSendTxData || !chain || !multiSendContract || !txsWithDetails) return
+
+    try {
+      const multiSendTransaction = await multiSendContract.connect(provider.getSigner()).multiSend(multiSendTxData)
+
+      txsWithDetails.forEach((tx) => {
+        createExistingTx(chain.chainId, safe.address.value, tx.txId, tx).then((safeTx) =>
+          txDispatch(TxEvent.MINING, { txId: tx.txId, txHash: multiSendTransaction.hash, tx: safeTx }),
```

Comment:
Oof, this is going to show lots of notifications.

Reply:
With the txHash passed and our change it should only display one notification